x-pack/plugins/observability_solution/observability_ai_assistant_app/server/functions/documentation.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { DocumentationProduct } from '@kbn/product-doc-common';
+import { FunctionVisibility } from '@kbn/observability-ai-assistant-plugin/common';
+import type { FunctionRegistrationParameters } from '.';
+
+export const RETRIEVE_DOCUMENTATION_NAME = 'retrieve_elastic_doc';
+
+export async function registerDocumentationFunction({
+  functions,
+  resources,
+  pluginsStart: { llmTasks },
+}: FunctionRegistrationParameters) {
+  const isProductDocAvailable = (await llmTasks.retrieveDocumentationAvailable()) ?? false;
+
+  functions.registerFunction(
+    {
+      name: RETRIEVE_DOCUMENTATION_NAME,
+      visibility: isProductDocAvailable
+        ? FunctionVisibility.AssistantOnly
```

Comment:
@elastic/obs-ai-assistant the new function I added to utilize the product documentation. 

Note that I had to add an async call to check if the product doc feature is available, thankfully the assistant framework already supports async function registration calls.

Reply:
Note to myself: I might want to add instructions too for better function discoverability by the LLM